DANS-KNAW/easy-deposit-api
# EASY-1751 mask authentication info in logging

Fixes EASY-1751

- [x] Process review comments.

#### When applied it will
* mask privacy sensitive info in the logging
* moved the request/response logging towards a potential library
* 

#### Where should the reviewer @DANS-KNAW/easy start?

Below examples of masked and not-masked variants of requests and responses. The masked variants are applied. Should less be masked or more elements of requests/response be logged?

https://github.com/DANS-KNAW/easy-deposit-api/blob/a969de90e568369f21a5754f335b5f627b393738/src/test/scala/nl.knaw.dans.easy.deposit/logging/RequestLogFormatterSpec.scala#L37
https://github.com/DANS-KNAW/easy-deposit-api/blob/a969de90e568369f21a5754f335b5f627b393738/src/test/scala/nl.knaw.dans.easy.deposit/logging/RequestLogFormatterSpec.scala#L44
https://github.com/DANS-KNAW/easy-deposit-api/blob/a969de90e568369f21a5754f335b5f627b393738/src/test/scala/nl.knaw.dans.easy.deposit/logging/ResponseLogFormatterSpec.scala#L34
https://github.com/DANS-KNAW/easy-deposit-api/blob/a969de90e568369f21a5754f335b5f627b393738/src/test/scala/nl.knaw.dans.easy.deposit/logging/ResponseLogFormatterSpec.scala#L40

#### How should this be manually tested?

* Run unit tests with log level "trace" at:
    https://github.com/DANS-KNAW/easy-deposit-api/blob/7d4caa48316f711f5daeebf91e126e2f6b75b075/src/test/resources/logback.xml#L15
* See [manual-tests] or deploy `easy-deposit-ui` too and use the browser.

[manual-tests]: https://github.com/DANS-KNAW/easy-deposit-api/blob/master/src/test/resources/manual-test/readme.md#manual-tests

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..
